code/main.c

```diff
@@ -16,7 +16,8 @@
 
 #include "io/io_save_hdf5.h"
 
-char bufz0[1000];
+#define MAX_BUFZ0_SIZE (3*MAX_STRING_LEN+20)
+char bufz0[MAX_BUFZ0_SIZE]; /* 3 strings + max 19 bytes for a number */
```

Comment:
Think you have the extra byte already accounted for - but might be good to change MAX_BUFZ0_SIZE to the `3*MAX_STRING_LEN + 19`, and then add the ` + 1` in the allocation

Reply:
Having a few (2 or 3) _unused_ bytes is low risk. Nobody's going to hide exploitable code there. The risk is people overwriting the buffer and jumping into RAM where they're not supposed to go. Caveat: this assumes that I understand the security issue here.

Reply:
I was not worried about the wasted bytes - simply thinking about being explicit about the calculation. The extra `+ 1` would show that that was the space for terminating NULL byte.